src/stdlib_experimental_io.f90

```diff
@@ -46,7 +46,7 @@ subroutine sloadtxt(filename, d)
 integer :: s
 integer :: nrow,ncol,i
 
-open(newunit=s, file=filename, status="old")
+open(newunit=s, file=filename, status="old", action="read")
```

Comment:
Is the `action` argument required to be used in `open`? I had to look it up what it means. The standard says "If this specifier is omitted, the default value is processor dependent.", which seems to imply that we have to use it, because if the processor dependent default value happens to be `read`, then we cannot write to the file. This seems to be a pretty bad design of the Fortran standard, similar to `status`, which you have to specify also, otherwise it is processor dependent.

Reply:
Agreed about the shortcoming of this in the standard. Either way it's good practice to

1. Indicate intent
2. Prevent accidental writing when you only should be reading

Reply:
`action=` is not a required parameter, but I think it's "safer" to use it to handle edge cases like use on network drives without write permission or other weird permissions. And just to help avoid writing a file if a handle variable is used inadvertently. I have not had any issues using `action=` everywhere.